hazelcast/src/main/java/com/hazelcast/cache/impl/ICacheRecordStore.java

```diff
@@ -448,4 +449,6 @@
      * @return <tt>true</tt> if wan replication is enabled for this record store, <tt>false</tt> otherwise
      */
     boolean isWanReplicationEnabled();
+
+    void instrument(NodeEngine nodeEngine);
```

Comment:
How about removing this from the interface and keeping this on the instance? Seems like something that not all implementations would want to have.